### PR TITLE
Misc fixes for darwin and android compilation on darwin

### LINKED ISSUE #39: MSG_NOSIGNAL is not defined on Mac OSX
MacOSX does not support the flag `MSG_NOSIGNAL`. This result in a compile failure on `Socket.cpp`

This is the failure log:
~~~
[  9%] Built target xmlserializer
[ 12%] Built target pfw_utility
[ 86%] Built target parameter
[ 87%] Building CXX object remote-processor/CMakeFiles/remote-processor.dir/Socket.cpp.o
/Users/makohoek/Documents/Development/parameter-framework/remote-processor/Socket.cpp: In member function 'bool CSocket::read(void*, uint32_t)':
/Users/makohoek/Documents/Development/parameter-framework/remote-processor/Socket.cpp:111:78: error: 'MSG_NOSIGNAL' was not declared in this scope
         int32_t iAccessedSize = ::recv(_iSockFd, &pucData[uiOffset], uiSize, MSG_NOSIGNAL);
                                                                              ^
/Users/makohoek/Documents/Development/parameter-framework/remote-processor/Socket.cpp: In member function 'bool CSocket::write(const void*, uint32_t)':
/Users/makohoek/Documents/Development/parameter-framework/remote-processor/Socket.cpp:143:78: error: 'MSG_NOSIGNAL' was not declared in this scope
         int32_t iAccessedSize = ::send(_iSockFd, &pucData[uiOffset], uiSize, MSG_NOSIGNAL);
                                                                              ^
make[2]: *** [remote-processor/CMakeFiles/remote-processor.dir/Socket.cpp.o] Error 1
make[1]: *** [remote-processor/CMakeFiles/remote-processor.dir/all] Error 2
make: *** [all] Error 2
~~~

CMake configuration:
https://gist.github.com/Makohoek/4bafe8fc72a6f0d1130a

See for more information about `MSG_NOSIGNAL` support in MacOSX
http://www.lists.apple.com/archives/macnetworkprog/2002/Dec/msg00091.html

### PR DESCRIPTION
**Pull request by dawagner**
Fixes and workarounds for darwin, mostly related to android. Closes #39.

### CODE REVIEW

**Comment by dawagner** [comment]
@OznOg @krocard please review.

File: bindings/python/Android.mk [hunk 1]
```diff
@@ -74,15 +79,29 @@ LOCAL_GENERATED_SOURCES := $(generated-sources-dir)/pfw_wrap.cxx $(generated-sou
 
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(generated-sources-dir)
 
+# Careful, we need to invoque the android python config not the host's one.
+# BTW, the intenal install directory of python is hardcoded to a dummy value,
+# thus we need to manually add the correct path to libs to the library list.
+ifeq ($(HOST_OS), linux)
```
Comment (krocard) [inline bindings/python/Android.mk]: no dummy value on darwin ?

File: bindings/python/Android.mk [hunk 2]
```diff
@@ -74,15 +79,29 @@ LOCAL_GENERATED_SOURCES := $(generated-sources-dir)/pfw_wrap.cxx $(generated-sou
...
+LOCAL_LDLIBS += $(shell python-config --ldflags)
+endif
+
 $(generated-sources-dir)/pfw_wrap.h: $(generated-sources-dir)/pfw_wrap.cxx
 
 # The PyPfw.py file is generated in the directory given by -outdir switch, thus
 # this directory must be put in the python path to be reachable
+$(generated-sources-dir)/pfw_wrap.cxx: HOST_ARCH := $(HOST_ARCH)
+$(generated-sources-dir)/pfw_wrap.cxx: HOST_OS := $(HOST_OS)
+$(generated-sources-dir)/pfw_wrap.cxx: HOST_LIBRARY_PATH := $(HOST_LIBRARY_PATH)
```
Comment (krocard) [inline bindings/python/Android.mk]: Not sure that this is needed as those variable should be immutable. Nevertheless, fool safe protection is not a bad idea in android build system.

File: remote-processor/Socket.cpp [hunk 3]
```diff
@@ -108,7 +108,7 @@ bool CSocket::read(void* pvData, uint32_t uiSize)
 
     while (uiSize) {
 
-        int32_t iAccessedSize = ::recv(_iSockFd, &pucData[uiOffset], uiSize, MSG_NOSIGNAL);
+        int32_t iAccessedSize = ::recv(_iSockFd, &pucData[uiOffset], uiSize, 0);
```
Comment (krocard) [inline remote-processor/Socket.cpp]: I guess there is no MSG_DEFAULT or similar (could not find anything like that in `man recv`) ?

File: remote-processor/Socket.cpp [hunk 4]
```diff
@@ -140,17 +140,19 @@ bool CSocket::write(const void* pvData, uint32_t uiSize)
 
     while (uiSize) {
 
-        int32_t iAccessedSize = ::send(_iSockFd, &pucData[uiOffset], uiSize, MSG_NOSIGNAL);
+        int32_t iAccessedSize = ::send(_iSockFd, &pucData[uiOffset], uiSize, 0);
```
Comment (krocard) [inline remote-processor/Socket.cpp]: I guess you are solving #39 with this pull request.

File: bindings/python/Android.mk [hunk 5]
```diff
@@ -74,15 +79,29 @@ LOCAL_GENERATED_SOURCES := $(generated-sources-dir)/pfw_wrap.cxx $(generated-sou
 
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(generated-sources-dir)
 
+# Careful, we need to invoque the android python config not the host's one.
```
Comment (dawagner) [inline bindings/python/Android.mk]: invoke

File: bindings/python/Android.mk [hunk 6]
```diff
@@ -74,15 +79,29 @@ LOCAL_GENERATED_SOURCES := $(generated-sources-dir)/pfw_wrap.cxx $(generated-sou
...
+# BTW, the intenal install directory of python is hardcoded to a dummy value,
```
Comment (dawagner) [inline bindings/python/Android.mk]: internal

File: bindings/python/Android.mk (hunk 2 above)
Comment (dawagner) [inline bindings/python/Android.mk]: I agree that we can remove this paragraph.

File: remote-processor/Socket.cpp (hunk 3 above)
Comment (dawagner) [inline remote-processor/Socket.cpp]: `flags` is an int and from `man recv`:

    The flags argument is formed by ORing one or more of the following values:

So, `0` is correct.

File: remote-processor/Socket.cpp (hunk 4 above)
Comment (dawagner) [inline remote-processor/Socket.cpp]: yes, as mentionned in the PR message.

**Comment by krocard** [comment]
Are you sure a SIGPIPE can be received ? I though that it was when reading a file descriptor after receiving EOF ?

File: remote-processor/Socket.cpp (hunk 3 above)
Comment (krocard) [inline remote-processor/Socket.cpp]: Agreed, but I wanted to know if there was a standard macro that represented this.

File: remote-processor/Socket.cpp (hunk 4 above)
Comment (krocard) [inline remote-processor/Socket.cpp]: Yes, :+1:

File: bindings/python/Android.mk (hunk 1 above)
Comment (krocard) [inline bindings/python/Android.mk]: Fixed in #124

File: bindings/python/Android.mk (hunk 2 above)
Comment (krocard) [inline bindings/python/Android.mk]: Fixed in #124

File: remote-processor/Socket.cpp (hunk 4 above)
Comment (krocard) [inline remote-processor/Socket.cpp]: Fixed in #124

File: bindings/python/Android.mk (hunk 5 above)
Comment (krocard) [inline bindings/python/Android.mk]: Fixed in #124

File: bindings/python/Android.mk (hunk 6 above)
Comment (krocard) [inline bindings/python/Android.mk]: Fixed in #124

**Comment by krocard** [comment]
Took owneship in #124 of this pull request.

File: bindings/python/Android.mk (hunk 2 above)
Comment (dawagner) [inline bindings/python/Android.mk]: *This* is #124. Did you mean #131 ?

**Pull request closed without merging**